packages/ckeditor5-engine/src/conversion/upcasthelpers.js

```diff
@@ -465,6 +465,11 @@ export function convertText() {
 				return;
 			}
 
+			// Do not auto-paragraph whitespaces.
```

Comment:
This breaks the following test: https://github.com/ckeditor/ckeditor5/blob/30286f77b39526fce2856b03b9be0ba4cc91d1c7/packages/ckeditor5-engine/tests/view/domconverter/whitespace-handling-integration.js#L110

We need to choose to either trim that nbsp in that case or correctly handle scenarios like:

```
<custom>x</custom>

<custom someattr></custom>
```

where the space (normal space) should be trimmed.

Unfortunately, since these are custom elements we can't easily make the decision on DOM->view conversion whether these are spaces between block or inline elements.

Additionally, unfortunately we lose the information whether this was an nbsp or normal space upon DOM->view conversion. Currently, all are normalized to normal spaces at this stage. if they weren't, the below code would actually correctly differentiate these situation.

Hence, we need to:

* either lose intentional nbsps between blocks (may happen in HTML generated by external tools)
* or make sure custom elements don't break things